### PR TITLE
IDS-872: Revive broken Crucible tests

### PR DESCRIPTION
**Pull request by andrew-uoa**
This PR reinstates the tests for the `Crucible` class, which were previously largely marked `xfail`.

The tests were previously based on patching request responses using the `Responses` package, but are now based on mocking out part of the `Overseer` class, which is used by `Crucible`. This was done because `Overseer` is a direct dependency of `Crucible`, while `Overseer` is responsible for lower-level request making via `MyTardisRESTFactory`. My thinking is that directly patching requests is too low-level for tests of this component.

Also includes some minor, highly-localised refactoring in `Crucible`, for readability.

### CODE REVIEW

**Review by uoa-noel — CHANGES_REQUESTED** [review]
Looks great! Thanks for changing the code as discussed. I have one more small suggestion, but looks good to merge otherwise.

File: tests/fixtures/fixtures_dataclasses.py [hunk 1]
```diff
@@ -456,6 +459,18 @@ def __call__(self, **kwargs: Any) -> T_co: ...
```
[the remark this answers is not in the document]
Comment (andrew-uoa) [reply tests/fixtures/fixtures_dataclasses.py:459]: As we discussed, in a future PR I will remove TestModelFactory in favour of something simpler

[the remark this answers is not in the document]
Comment (uoa-noel) [reply tests/fixtures/fixtures_dataclasses.py:459]: As discussed this change will be dealt with in a separate PR, so will resolve it in this PR!

**Review by uoa-noel — APPROVED** [review]
LGTM! :+1:

**Pull request merged**